File: src/main/java/no/nav/syfo/api/intern/domain/RSGodkjentPlan.java
package no.nav.syfo.api.intern.domain;

import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@Data
@Accessors(fluent = true)
public class RSGodkjentPlan {
    public LocalDateTime opprettetTidspunkt;
    public RSGyldighetstidspunkt gyldighetstidspunkt;
    public boolean tvungenGodkjenning;
    public LocalDateTime deltMedNAVTidspunkt;
    public boolean deltMedNAV;
    public LocalDateTime deltMedFastlegeTidspunkt;
    public boolean deltMedFastlege;
    public String dokumentUuid;
}

File: src/main/kotlin/no/nav/syfo/api/v3/domain/oppfoelgingsdialog/RSGodkjentPlan.kt
package no.nav.syfo.api.v3.domain.oppfoelgingsdialog

import java.time.LocalDateTime

data class RSGodkjentPlan(
    var opprettetTidspunkt: LocalDateTime? = null,
    var gyldighetstidspunkt: RSGyldighetstidspunkt? = null,
    var tvungenGodkjenning: Boolean = false,
    var deltMedNAVTidspunkt: LocalDateTime? = null,
    var deltMedNAV: Boolean = false,
    var deltMedFastlegeTidspunkt: LocalDateTime? = null,
    var deltMedFastlege: Boolean = false,
    var dokumentUuid: String? = null
)
